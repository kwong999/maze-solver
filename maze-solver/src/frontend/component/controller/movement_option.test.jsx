import React from 'react';
import MovementOption from './movement_option';
import renderer from 'react-test-renderer';

const createComponent = str => {
  return renderer.create(
    <MovementOption
      movement={str}
      changeMovement={() => { }}
    />
  );
}

describe('MovementOption', () => {
  test('have class name "all-direction"', () => {
    const component = createComponent("All Direction");
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('not have class name "all-direction"', () => {
    const component = createComponent("No Diagonal");
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});