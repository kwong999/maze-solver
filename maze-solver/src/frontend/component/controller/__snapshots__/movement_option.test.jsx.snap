// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovementOption have class name "all-direction" 1`] = `
<div
  className="movement-option maze-action"
>
  <div>
    <p>
      Movement: 
    </p>
    <button
      className="all-direction"
      onClick={[Function]}
    >
      All Direction
    </button>
  </div>
</div>
`;

exports[`MovementOption not have class name "all-direction" 1`] = `
<div
  className="movement-option maze-action"
>
  <div>
    <p>
      Movement: 
    </p>
    <button
      className=""
      onClick={[Function]}
    >
      No Diagonal
    </button>
  </div>
</div>
`;
